### PR TITLE
Add cuda tests

### PR DESCRIPTION
**Pull request by lukasm91**
Adds a specific test for OpenACC support in array wrappers. Tests only compilation, but not execution because obviously, it's nearly impossible to have only one toolchain that compiles fortran and C++ successfully.

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Review by havogt — APPROVED** [review]

File: cmake/cpp_bindgen.cmake.in [hunk 1]
```diff
@@ -110,7 +110,6 @@ function(cpp_bindgen_add_library target_name)
 
     add_library(${target_name} ${ARG_SOURCES})
     target_link_libraries(${target_name} PRIVATE cpp_bindgen_generator)
-    # target_include_directories(${target_name} PRIVATE ${PROJECT_SOURCE_DIR}/include) #TODO probably wrong
```
Comment (lukasm91) [inline cmake/cpp_bindgen.cmake.in:113]: I agree, we should remove this (it should already be part of `cpp_bindgen_generator` anyway I think.

**Review by lukasm91 — COMMENTED** [review]
I would remove the TODO.

**Review by havogt — APPROVED** [review]

**Pull request merged**